packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -997,11 +997,13 @@ function updateClassInstance(
 
   cloneUpdateQueue(current, workInProgress);
 
-  const oldProps = workInProgress.memoizedProps;
-  instance.props =
+  const unresolvedOldProps = workInProgress.memoizedProps;
```

Comment:
This says they're unresolved. But later we do this:

```js
instance.props = workInProgress.memoizedProps = newProps;
```

So next time we *would* have resolved props in `memoizedProps`? I don't quite follow the logic.

Reply:
Since we reset `workInProgress.memoizedProps` to `workInProgress.pendingProps` right after begin work, `workInProgress.memoizedProps` is always unresolved when we begin work.
https://github.com/facebook/react/blob/e69ca310ea56b6f2f510dd463902405285953432/packages/react-reconciler/src/ReactFiberWorkLoop.js#L1501

Reply:
Then can we assign `instance.props` only? Is there a reason for `workInPorgress.memoizedProps` assignment? Sorry I'm being dense, it's been a while since I looked a tthis.

Reply:
Never mind. But unfortunately we have to keep their values in sync. [Here](https://github.com/facebook/react/pull/18491#discussion_r403638636) is the reason.

Reply:
I still don't think it makes sense to assign either of them.

In the sCU branch, `instance.props` reassignment makes sense because the passed props actually *did* change. We want the most recent values inside of them to be accessible to the lifecycle methods, even if sCU "lies" about them not having changed.

In this branch, we've verified props are `===`. So the only reason they "changed" is because we resolved default props. That's not a compelling reason to give our components a different props object.

Reply:
If you change the warning condition to `if (shouldUpdate && inst.props !== nextProps) {` then the warning won't fire. It will also make sense because *the render wasn't called* — so the warning message would have been misleading anyway.

Reply:
```
instance.props = workInProgress.memoizedProps = newProps;
```
This only gives our components a different props object when we've actually resolved default props. But in that case, components are already given a new props object.
https://github.com/facebook/react/blob/2dddd1e00cdadad1783aa55ac65a28634590e9c1/packages/react-reconciler/src/ReactFiberClassComponent.js#L1001-L1004
I think it's ok to remove this assignment if keeping `workInProgress.memoizedProps` and `instance.props` stay the same until beginWork finishes is not a **requirement**. And that seems true now. So let's change the warning condition instead.